### PR TITLE
Set minimum supported Matrix 1.1 version (drop legacy r0 versions)

### PR DESCRIPTION
**Pull request by turt2live**
WIP

Fixes https://github.com/vector-im/element-web/issues/16876
**Requires https://github.com/matrix-org/matrix-react-sdk/pull/9819**

Prerequisites: 
* [x] **Requires https://github.com/matrix-org/matrix-js-sdk/pull/3005**
* [x] **Requires https://github.com/matrix-org/matrix-js-sdk/pull/3003**
* [x] **Requires https://github.com/matrix-org/matrix-react-sdk/pull/9818**
* [x] **Requires https://github.com/vector-im/element-web/pull/24086**
* [x] **Requires https://github.com/matrix-org/matrix-js-sdk/pull/3006**

---
Here's what your changelog entry will look like:

## 🚨 BREAKING CHANGES
 * Set minimum supported Matrix 1.1 version (drop legacy r0 versions) ([\#3007](https://github.com/matrix-org/matrix-js-sdk/pull/3007)). Fixes vector-im/element-web#16876.

### CODE REVIEW

**Comment by andybalaam** [comment]
@turt2live is this still ongoing?

**Comment by turt2live** [comment]
Not by me, but is increasingly critical work the team will need to pick up.

**Comment by turt2live** [comment]
I don't recall, sorry. This will need investigation from the team.

**Comment by turt2live** [comment]
this PR should also probably update the README text to say there's now a minimum version

File: src/http-api/prefix.ts [hunk 1]
```diff
@@ -16,11 +16,7 @@ limitations under the License.
 
 export enum ClientPrefix {
     /**
-     * A constant representing the URI path for release 0 of the Client-Server HTTP API.
-     */
-    R0 = "/_matrix/client/r0",
-    /**
-     * A constant representing the URI path for the legacy release v1 of the Client-Server HTTP API.
+     A constant representing the URI path for Client-Server API endpoints versioned at v1.
```
Comment (germain-gg) [inline src/http-api/prefix.ts:19]:
```suggestion
     * A constant representing the URI path for Client-Server API endpoints versioned at v1.
```

**Review by germain-gg — APPROVED** [review]
lgtm

File: src/http-api/prefix.ts (hunk 1 above)
Comment (t3chguy) [reply src/http-api/prefix.ts:19]: Argh, need to figure out why WebStorm strips the leading `*` when copy-pasting into a jsdoc block, thanks

**Pull request merged**